src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,185 @@
+package duke.task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.util.DukeException;
+import duke.util.Parser;
+
+/**
+ * A TaskList which stores Tasks.
+ */
+public class TaskList {
+    // Messages
+    private static final String OUT_OF_BOUNDS_TASK = "Could not find task. Check the task number again?";
+    private static final String NUMBER_OF_TASKS_MESSAGE = "Now you have %d %s in the list.";
+    private static final String ADD_TASK_MESSAGE = "Got it. I've added this task:\n  %s\n" + NUMBER_OF_TASKS_MESSAGE;
+    private static final String REMOVE_TASK_MESSAGE = "Noted. I've removed this task:\n %s\n" + NUMBER_OF_TASKS_MESSAGE;
+    private static final String ERROR_SAVING_MESSAGE = "Error reading taskLst. Symbol not found.";
+    private static final String NO_TASK_FOUND_MESSAGE = "Unfortunately no tasks with that name are found";
+    private static final String MATCHING_TASKS_MESSAGE = "Here are the matching tasks in your list:\n";
+
+    private final List<Task> taskArr;
+
+    /**
+     * Constructor for TaskList
+     */
+    public TaskList(String taskList) throws DukeException {
+        this.taskArr = new ArrayList<>();
+        String[] taskArrStr = taskList.split("\n");
+        for (String task : taskArrStr) {
+            this.addSavedTask(task);
+        }
+    }
+
+    public TaskList() {
+        this.taskArr = new ArrayList<>();
+    }
+
+    // Nouns for plural and singular
+    private String taskWord() {
+        return this.size() <= 1 ? "task" : "tasks";
+    }
+
+
+    /**
+     * The size of the array storing the Tasks.
+     * @return The size of the array storing the Tasks.
+     */
+    public int size() {
```

Comment:
This could also be changed to getSize so that your methods are all consistent to the naming conventions (i.e. methods named as verbs)?

Reply:
Thanks for the comments! I'll work on addressing the issues you pointed out.